primitives/state-machine/src/in_memory_backend.rs

```diff
@@ -23,22 +23,36 @@ use crate::{
 use codec::Codec;
 use hash_db::Hasher;
 use sp_core::storage::{ChildInfo, StateVersion, Storage};
-use sp_trie::{empty_trie_root, LayoutV1, MemoryDB};
+use sp_trie::{empty_trie_root, GenericMemoryDB, HashKey, KeyFunction, LayoutV1, MemoryDB};
 use std::collections::{BTreeMap, HashMap};
 
 /// Create a new empty instance of in-memory backend.
-pub fn new_in_mem<H: Hasher>() -> TrieBackend<MemoryDB<H>, H>
+///
+/// It will us [`HashKey`] to store the keys internally.
+pub fn new_in_mem_hash_key<H: Hasher>() -> TrieBackend<MemoryDB<H>, H>
```

Comment:
```suggestion
pub fn new_in_mem_hash_key<H>() -> TrieBackend<MemoryDB<H>, H>
```

Reply:
I can propose to rename to `new_in_mem_unprefixed` or `new_in_mem_no_prefix`. But it does not help that much (still need to get an idea for the keyfunction prefixing usage).